From a31bd8391c5bb4c34717ec7039092e828405c298 Mon Sep 17 00:00:00 2001
From: Andy Duan <fugang.duan@nxp.com>
Date: Mon, 14 May 2018 20:46:31 +0800
Subject: [PATCH 3800/5242] MLK-18313 ARM64: dts: imx8mq-evk-pcie1-m2: disable
 pcie0 port

commit  562d5ada11bd59110d64413bbffa10696073e3f7 from
https://source.codeaurora.org/external/imx/linux-imx.git

When wifi driver switch to QCA CLD from ATH10K, there have one known issue:
   - QCA CLD driver only support ONE instance.
So it has to disable pcie0 port.

Reviewed-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8mq-evk-pcie1-m2.dts |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-pcie1-m2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-pcie1-m2.dts
index feae708..6b01476 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-pcie1-m2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-pcie1-m2.dts
@@ -61,6 +61,12 @@
  *	&usdhc2 {
  *		status = "disabled";
  *	};
+ *
+ *
+ * 3. When wifi driver switch to QCA CLD from ATH10K, there have two known issues:
+ *	- QCA CLD driver only support ONE instance.
+ *	- QCA CLD driver don't support PCIe MSI feature.
+ *    So here it has to disable pcie0 port.
  */
 
 #include "fsl-imx8mq-evk.dts"
@@ -70,3 +76,7 @@
 		reset-gpios = <&gpio3 5 GPIO_ACTIVE_HIGH>;
 	};
 };
+
+&pcie0{
+	status = "disabled";
+};
-- 
1.7.9.5

